great_expectations/rule_based_profiler/rule/rule.py

```diff
@@ -79,27 +81,69 @@ def generate(
 
         return expectation_configurations
 
+    def to_dict(self) -> dict:
+        parameter_builders_as_dicts: Optional[List[dict]] = None
+        parameter_builders: Optional[
+            Dict[str, ParameterBuilder]
+        ] = self._get_parameter_builders_as_dict()
+        parameter_builder: ParameterBuilder
+        if parameter_builders is not None:
+            parameter_builders_as_dicts = [
+                parameter_builder.to_dict()
+                for parameter_builder in list(parameter_builders.values())
+            ]
+
+        expectation_configuration_builders_as_dicts: Optional[List[dict]] = None
+        expectation_configuration_builders: Optional[
+            Dict[str, ExpectationConfigurationBuilder]
+        ] = self._get_expectation_configuration_builders_as_dict()
+        expectation_configuration_builder: ExpectationConfigurationBuilder
+        if expectation_configuration_builders is not None:
+            expectation_configuration_builders_as_dicts = [
+                expectation_configuration_builder.to_dict()
+                for expectation_configuration_builder in list(
+                    expectation_configuration_builders.values()
+                )
+            ]
```

Comment:
I appreciate verbosity but this is really hard to read in my opinion.

Also, I don't think we need `list` here or above (you can use the iterator as is)

Reply:
@cdkini Indeed, `list` is not needed (deleted in the next commit).  I would like to keep this verbose style in for now.  Thanks.